models/ArticleManager.php

```diff
@@ -0,0 +1,377 @@
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
+
+            if(empty($errors)){
+                ArticleManager::post($title,$chapo,$content,$posted);
+                if(!empty($_FILES['image']['name'])){
+                    Articlemanager::post_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }else{
+                   header("Location:/liste-de-tous-les-articles");
+                }
+            }
+        }
+        else
+            $errors =[];
+        
+        return $errors;
+    }
+    
+    function postback_verification()
+    {
+            if(filter_has_var(INPUT_POST, 'delete')){
+            ArticleManager::delete_article_comments();
+            ArticleManager::delete_article();
+        }
+
+        if(filter_has_var(INPUT_POST, 'submit')){
+            if(filter_has_var(INPUT_POST, 'title')){
+                $title = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'title')));
+            }
+            if(filter_has_var(INPUT_POST, 'chapo')){
+                $chapo = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'chapo')));
+            }
+            if(filter_has_var(INPUT_POST, 'content')){
+                $content = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'content')));
+            }
+            $posted = filter_input(INPUT_POST, 'public') ? "1" : "0";
+            
+            $errors = [];
+
+            if(empty($title) || empty($content) || empty($chapo)){
+                $errors['empty'] = "Veuillez remplir tous les champs svp";
+            }
+            if(strlen($title) < 5){
+                $errors['title'] = "Votre message doit contenir au moins 5 caractères.";
+            }
+            if(strlen($chapo) < 5){
+                $errors['chapo'] = "Votre chapô doit contenir au moins 5 caractères.";
+            }
+            if(strlen($content) < 5){
+                $errors['content'] = "Votre article doit contenir au moins 5 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+  
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable.";
+                }
+            }
+            
+            if(empty($errors)){
+                ArticleManager::edit($title,$chapo,$content,$posted, filter_input(INPUT_GET, 'id'));
+             
+                if(!empty($_FILES['image']['name']))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_FILES Superglobal detected.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)